osf_tests/test_user.py

```diff
@@ -219,6 +219,34 @@ def test_merged_user_with_two_account_on_same_project_with_different_visibility_
         assert project.get_visible(user) is True
         assert project.is_contributor(user2) is False
 
+    def test_merge_preprints(self, user):
+        user2 = AuthUserFactory()
+
+        user_is_creator = PreprintFactory(creator=user2)
+
+        contrib_not_creator = PreprintFactory()
+        contrib_not_creator.add_contributor(user)
+
+        # if not handled well this can throw an IntegrityError
+        both_users_are_contribs = PreprintFactory()
+        both_users_are_contribs.add_contributor(user)
+        both_users_are_contribs.add_contributor(user2)
+
+        user2.merge_user(user)
+
+        qs = user.preprints.all()
+        assert qs.count() == 0
+
+        qs = user2.preprints.all()
+        assert qs.count() == 3
+
+        qs = user2.preprints.filter(creator=user2)
+        assert qs.count() == 1
```

Comment:
Also please add tests for permission/bibliographic information in both cases, 1) where both users originally belonged to the preprint 2) only the second user belonged to the preprint.

In the first case, the highest permission and bibliographic info needs to be transferred, in the second, bib/permission info are transferred to the user we're keeping, and perms are removed from the merged user.